data-raw/02_process_mars_gribs_to_COGS.py

```diff
@@ -0,0 +1,88 @@
+import os
+import tempfile
+from pathlib import Path
+
+import pandas as pd
+import xarray as xr
+
+from src.blob_utils import (
+    download_file,
+    list_blobs,
+    load_env_vars,
+    upload_file,
+)
+
+SAS_TOKEN, CONTAINER_NAME, STORAGE_ACCOUNT = load_env_vars()
+
+grib_files = list_blobs(
+    sas_token=SAS_TOKEN,
+    container_name=CONTAINER_NAME,
+    storage_account=STORAGE_ACCOUNT,
+    name_starts_with="ds-aa-afg-drought/raw/",
+)
+
+grib_files = [file for file in grib_files if file.endswith(".grib")]
+
+for blob_name in grib_files:
+    print(f"name: {os.path.basename(blob_name)}")
+
+    # Create a temporary file
+    with tempfile.TemporaryDirectory() as td:
+        temp_base_path = os.path.basename(blob_name)
+        td = Path(td)
+        tf = td / temp_base_path
+        # Download the file
+        download_file(
+            sas_token=SAS_TOKEN,
+            container_name=CONTAINER_NAME,
+            storage_account=STORAGE_ACCOUNT,
+            blob_path=blob_name,
+            local_file_path=tf,
+        )
+
+        # Open the file with xarray
+        ds = xr.open_dataset(
+            tf,
+            engine="cfgrib",
+            drop_variables=["surface", "values"],
+            backend_kwargs=dict(time_dims=("time", "forecastMonth")),
+        )
+
+        ds["longitude"] = (ds["longitude"] + 180) % 360 - 180
+        ds["precip"] = (
+            ds["tprate"] * ds["time"].dt.days_in_month * 24 * 3600 * 1000
+        )
+        ds = ds.drop_vars("tprate")
+        da = ds["precip"]
+        da_mean = da.mean(dim=["number"])
+
+        # loop through pub dates & COGS
+        pub_dates = da_mean.time.values
+        forecast_months = da_mean.forecastMonth.values
+
+        # give a CRS
+        da_mean = da_mean.rio.write_crs("EPSG:4326", inplace=False)
```

Comment:
Yeah this is the better way to do this, I've found that `inplace=True` doesn't actually work.

Reply:
ha good to know it doesn't actually work! maybe should make an issue of it in rio?